src/Facebook/InstantArticles/Validators/Type.php

```diff
@@ -362,4 +362,28 @@ private static function throwNotWithinException($value, $universe)
             " to be within this universe of values \n====[\n".$universe_str."]===="
         );
     }
+
+    /**
+     * Checks the thext if it is empty.
+     * "" => true
+     * "    " => true
+     * "\n" => true
+     * "a" => false
+     * "  a  " => false
+     *
+     * @param string $text The text that will be checked.
+     * @return true if empty, false otherwise.
+     */
+    public static function isTextEmpty($text)
+    {
+        if (!isset($text) || $text === null || !self::is($text, self::STRING)) {
+            return true;
+        }
+        // Stripes empty spaces, &nbsp;, <br/>, new lines
+        $text = strip_tags($text);
+        $text = preg_replace("/[\r\n\s]+/", "", $text);
+        $text = str_replace("&nbsp;", "", $text);
```

Comment:
Not sure where the `$text` param comes from, but for me in my attempts to strip whitespace for checking if a `DOMElement::nodeValue` was empty, I needed to do:

```php
$trimmed = trim($child->nodeValue, " \t\n\r\0\x0B\xC2\xA0");
if (!empty($trimmed)) {
  return FALSE;
}
```